streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -1292,7 +1275,14 @@ public void shouldThrowStreamsExceptionWhenFetchCommittedFailed() {
     public void shouldThrowIfCommittingOnIllegalState() {
         task = createStatelessTask(createConfig(false, "100"), StreamsConfig.METRICS_LATEST);
 
-        assertThrows(IllegalStateException.class, task::commit);
+        assertThrows(IllegalStateException.class, task::prepareCommit);
+    }
+
+    @Test
+    public void shouldThrowIfPostCommittingOnIllegalState() {
```

Comment:
Not introduced in this PR: could we add test checking CLOSED state should not commit as well?

Also checking SUSPENDED state close-call is no-op.

Reply:
Ack

Reply:
> Also checking SUSPENDED state close-call is no-op.

But in `close()` if state is suspended we might still wipe out the state store -- it's not a no-op